consensus/consensus-types/src/accumulator_extension_proof.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use failure::ensure;
+#[cfg(test)]
+use libra_crypto::hash::{CryptoHash, TestOnlyHasher, ACCUMULATOR_PLACEHOLDER_HASH};
+use libra_crypto::hash::{CryptoHasher, HashValue};
+#[cfg(test)]
+use libra_types::proof::TestAccumulatorInternalNode;
+use libra_types::proof::{accumulator::InMemoryAccumulator, definition::LeafCount};
+use serde::{Deserialize, Serialize};
+use std::marker::PhantomData;
+
+/// A proof that first verifies that establishes correct computation of the root and then
+/// returns the new tree to acquire a new root and version. Note: this is used internally by
+/// VoteProposal hence why it exists within consensus-types andd not libra-types.
+/// @TODO This should contain AccumulatorConsistencyProof once that is code complete
+#[derive(Deserialize, Serialize)]
+pub struct AccumulatorExtensionProof<H> {
+    /// Represents the roots of all the full subtrees from left to right in the original accumulator.
+    frozen_subtree_roots: Vec<HashValue>,
+    /// The total number of leaves in original accumulator.
+    num_leaves: LeafCount,
+    /// The values representing the newly appended leaves.
+    leaves: Vec<HashValue>,
```

Comment:
Maybe name this `new_leaves`.

Reply:
Better than the old name that was "subtrees" :O